### DISCUSSION TITLE
Support for jitpack

### DISCUSSION THREAD

**Discussion by yschimke** [jreleaser/jreleaser#629] [Ideas]
I use jitpack for easy publishing of low popularity artifacts.  As part of my publishing, jitpack should automatically build the artifacts when first used.

It would be great if jreleaser would trigger the build (there are webhooks for this, so it's redundant, but keeps it in one place), and possibly check if the build has passed or report the error if it fails.

API here - https://jitpack.io/docs/API/

**Comment by yschimke** [comment]
This is triggered because my changes to support jreleaser broke jitpack builds and I didn't notice.

**Reply by aalmiray** [reply]
When exactly in the release workflow were these artifacts published?
Was it when committing to git? Or was it when creating a Git release using the `jreleaserRelease` task?

**Reply by yschimke** [reply]
Creating the tag (before jreleaser gets involved) is enough to trigger it at some point*.

But jitpack either requires a webhook setup in github, or for someone to pull the version, meaning the first gradle reload in Intellij often takes minutes.

**Reply by aalmiray** [reply]
The more I think about this I don't see how this would be feasible for medium to large projects.

If a jitpack build should be triggered upon creating a release then JReleaser **could** potentially trigger a jitpack build. If this feature were added then the trigger would work from anywhere (local and CI) which is one of JReleaser's strengths. In the mean time you could use a CI hook (in this case GitHub Actions is the most likely option) to trigger the jitpack build. The downside of this is relying on CI to push the trigger.

Figuring out if a jitpack build was successful is trickier. Should it happen as part of the current release? If so the executing command must wait til the build has an outcome. This requires polling the jitpack API, there must be a maximum number of retries, timeouts, etc. For a small build this probably is not much of a problem (a couple of extra minutes added to the release as the build must be executed twice! Once in the current release, twice in jitpack) but for a medium to large project this would be a nightmare. It would be much better if jitpack offered push notifications but then _who_ should receive the notification if the release process has already finished before the jitpack build is done? Adding a JReleaser service on a remote host to handle this particular behavior doesn't look good at this moment.

**Reply by yschimke** [reply]
Thanks, makes sense.  I think the theme of my suggestions is that I'd like jreleaser to take care of my releases end to end. From triggering the release, through to ensuring everything completes.  But I guess the end game of that is a hosted service to start a release and that monitors the results.  While people come to jreleaser because it integrates into their existing workflows.
